### PR TITLE
feat(server): add originalFileName to asset table

### PR DESCRIPTION
**Pull request by alextran1502**
Add the original file name to the Asset table to use when we rerun failed jobs.

## Test
- [x] From `main` branch, update the assets with name, create three conditions of `imageName` valid on Exif table, `imageName` is NULL and `imageName` is an empty string. Switch to this branch and run migration
- [x] Seach metadata using image name
- [x] Refactor and check the web display image name correctly.
- [x] Refactor and check mobile app display the image name correctly.

### CODE REVIEW

**Comment by bo0tzz** [comment]
Why not just move this field to the asset table entirely, and remove it from the exif? I think that makes more sense as the field does not come from the exif data.

**Comment by alextran1502** [comment]
> Why not just move this field to the asset table entirely, and remove it from the exif? I think that makes more sense as the field does not come from the exif data.

We can certainly create a migration file for it

File: server/libs/infra/src/entities/asset.entity.ts [hunk 1]
```diff
@@ -87,6 +87,9 @@ export class AssetEntity {
   @Column({ nullable: true })
   livePhotoVideoId!: string | null;
 
+  @Column({ type: 'varchar', nullable: true })
+  originalFileName!: string | null;
```
Comment (jrasm91) [inline server/libs/infra/src/entities/asset.entity.ts:91]: This should not be nullable

File: server/libs/infra/src/migrations/1681144628393-AddOriginalFileNameToAssetTable.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class AddOriginalFileNameToAssetTable1681144628393 implements MigrationInterface {
+    name = 'AddOriginalFileNameToAssetTable1681144628393'
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "assets" ADD "originalFileName" character varying`);
```
Comment (jrasm91) [inline server/libs/infra/src/migrations/1681144628393-AddOriginalFileNameToAssetTable.ts:7]: We can probably do a migration here to update original filename from the exif table.

**Review by jrasm91 — COMMENTED** [review]
We should update the jobs to use this field and also save it here when an asset is created.

**Comment by truongsinh** [comment]
Just a quick note. Removing a column/field is a breaking change. if it is used by less-than-3-month-old releases, it might not let the user to downgrade to older version if they use Docker compose. Probably we need a dedicated section in https://immich.app/docs/install/docker-compose talking about DB schema change once Immich is stable.

**Comment by bo0tzz** [comment]
For the moment, we're not concerned at all about breaking changes. Once we go stable we do indeed need to establish a clear system for those sorts of things.

**Pull request merged**